internal/core/item_tree.rs

```diff
@@ -81,6 +78,9 @@ pub enum ItemTreeNode<T> {
 
         /// The index of the parent item (not valid for the root)
         parent_index: u32,
+
+        /// The index in the extra item_array
+        item_array_index: u32,
```

Comment:
I wonder if this is really needed and the extra_item_array  could just contain gap for the dynamic tree.

Reply:
Considering that the TreeNode gets smaller in any case with this change, I was not overly concerned about adding this index. I did not want to bother with the FFI for an `Option<...>`... 

This does give us some extra flexibility in that we could e.g. sort the item_array into traversal order to speed up rendering or event delivery.